### PR TITLE
unpin cftime

### PR DESCRIPTION
**Pull request by mathause**
I think the cftime problems should be fixed after the release of [v1.0.4.2](https://github.com/Unidata/cftime/releases/tag/v1.0.4.2rel)

 #3434

### CODE REVIEW

File: ci/requirements/py36-min-all-deps.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ dependencies:
   - cartopy=0.17
   - cdms2=3.1
   - cfgrib=0.9
-  - cftime=1.0.3  # FIXME need 1.0.5 (not released yet); 1.0.4 is broken
+  - cftime
```
Comment (spencerkclark) [inline ci/requirements/py36-min-all-deps.yml]: I think we may want to keep the version in this particular file pinned at `1.0.3` as it represents the new minimum version of cftime we will test against for the time being; xref: https://github.com/pydata/xarray/pull/3431#discussion_r337606766.
```suggestion
  - cftime=1.0.3
```

**Comment by mathause** [comment]
Ahh, yes of course... it's getting late.

File: ci/requirements/py36-min-all-deps.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ dependencies:
   - cartopy=0.17
   - cdms2=3.1
   - cfgrib=0.9
-  - cftime=1.0.3  # FIXME need 1.0.5 (not released yet); 1.0.4 is broken
+  - cftime=1.0.3
```
Comment (crusaderky) [inline ci/requirements/py36-min-all-deps.yml]: Could you change to 1.0, as per our policy of not using patch versions?

File: ci/requirements/py36-min-all-deps.yml (hunk 1 above)
Comment (crusaderky) [reply ci/requirements/py36-min-all-deps.yml]: As per our versions policy, we only support the latest available patch version
http://xarray.pydata.org/en/stable/installing.html#minimum-dependency-versions

**Comment by mathause** [comment]
Are the release notes fine, I wasn't sure what you are expecting.

**Comment by crusaderky** [comment]
Looks good to me

**Comment by max-sixty** [comment]
Thanks @mathause !

**Pull request merged**